packages/symbiosis/src/Symbiosis.test.ts

```diff
@@ -1,138 +1,79 @@
-        })
+  describe('should pass filter with valid transactions', () => {
+    passingTestCases.forEach((testCase) => {
+      const { transaction, description, params } = testCase
+      test(description, async () => {
+        const filter = await bridge(params)
         expect(apply(transaction, filter)).to.be.true
       })
     })
   })
-  describe('should not pass filter with invalid parameters', () => {
-    test('when sourceChainId is incorrect', async () => {
-      const { transaction, tokenAddress, recipient } = controlTransaction
 
-      const filter = await bridge({
-        sourceChainId: 1, // 42161
-        destinationChainId: 5000,
-        tokenAddress,
-        amount: GreaterThanOrEqual(10000n),
-        recipient,
-      })
-      expect(apply(transaction, filter)).to.be.false
-    })
-    test('when bridge contract address is incorrect', async () => {
-      const { transaction, tokenAddress, recipient } = controlTransaction
-
-      const filter = await bridge({
-        sourceChainId: 42161,
-        destinationChainId: 5000,
-        contractAddress: '0x1DCfbC3fA01b2a86bC3a3f43479cCe9E8D438Adc',
-        tokenAddress,
-        amount: GreaterThanOrEqual(10000n),
-        recipient,
-      })
-      expect(apply(transaction, filter)).to.be.false
-    })
-    FAILING_TEST_TRANSACTIONS.forEach((testTransaction) => {
-      test(testTransaction.description, async () => {
-        const {
-          transaction,
-          destinationChainId,
-          tokenAddress,
-          amount,
-          recipient,
-        } = testTransaction
-
-        const filter = await bridge({
-          sourceChainId: transaction.chainId,
-          destinationChainId,
-          tokenAddress,
-          amount: GreaterThanOrEqual(amount),
-          recipient,
-        })
+  describe('should not pass filter with invalid transactions', () => {
+    failingTestCases.forEach((testCase) => {
+      const { transaction, description, params } = testCase
+      test(description, async () => {
+        const filter = await bridge(params)
         expect(apply(transaction, filter)).to.be.false
       })
     })
   })
-  describe('control transaction should pass filter', () => {
-    test(controlTransaction.description, async () => {
-      const {
-        transaction,
-        destinationChainId,
-        tokenAddress,
-        amount,
-        recipient,
-      } = controlTransaction
 
-      const filter = await bridge({
-        sourceChainId: transaction.chainId,
-        destinationChainId,
-        tokenAddress,
-        amount: GreaterThanOrEqual(amount),
-        recipient,
+  describe('should return a valid list of tokens for each supported chain', async () => {
+    const chainIdArray = await getSupportedChainIds()
+    chainIdArray.forEach((chainId) => {
+      test(`for chainId: ${chainId}`, async () => {
+        const tokens = await getSupportedTokenAddresses(chainId)
+        const addressRegex = /^0x[a-fA-F0-9]{40}$/
```

Comment:
[nit] I would tend to use viem's `isAddress` to check if a string is a valid address
https://viem.sh/docs/utilities/isAddress.html#isaddress

BUT it does actually use this regex!
https://github.com/wevm/viem/blob/main/src/utils/address/isAddress.ts

Reply:
Learned something new! Awesome, I had no idea about this. Will change that out and start using it in all new plugins.